src/components/toast/toast.tsx

```diff
@@ -0,0 +1,205 @@
+import {
+  Component,
+  ComponentInterface,
+  h,
+  Prop,
+  Method,
+  Element,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
+
+export type ActionType = "button" | "icon";
+
+export type VariantType = "system" | "error" | "success" | "warning";
+
+export type ButtonActionType = "close" | "custom";
```

Comment:
Other solution would be to use `const enum`, It would avoid using magic string over the code